src/popupmenu.c

```diff
@@ -737,9 +737,11 @@ pum_redraw(void)
 						&& pum_width - totwidth < cells;
```

Comment:
I think you can move the `next_isempty` assignment below somewhere earlier and use this:
```suggestion
					&& pum_width - totwidth < cells + !next_isempty;
```

And then you can remove the condition below.

Reply:
this may not work 19 and 20 failed this works 

```C
			    int		truncated = pum_width == p_pmw
					    && ((pum_width - totwidth < cells)
						|| (pum_width - totwidth - cells - 1 == 0));
```

Reply:
i tried it test case failed this is work 

```C
			    int		truncated = pum_width == p_pmw
						&& ((pum_width - totwidth < cells)
						|| (!next_isempty && pum_width - totwidth - cells - 1 == 0));
```

Reply:
Oh, `+ next_isempty ? 0 : 2,` then?

Reply:
yes 2 space padding